simson/common/data_transformations.py

```diff
@@ -67,28 +67,21 @@ def extrapolate_stock(
 
 def scale_parameter_to_future(historic_parameter : NamedDimArray, scaler : NamedDimArray) -> NamedDimArray:
     """Scale parameter to future."""
-    dims = historic_parameter.dims.union_with(scaler.dims)
-
-    scaling_year = historic_parameter.dims.dim_list[0].items[-1]
     n_historic = historic_parameter.shape[0]
+    dims = historic_parameter.dims.union_with(scaler.dims)
 
-    transform_t_thist = get_subset_transformer(dims=dims, dim_letters=('t', 'h'))
     future_parameter = get_new_array(dims, dim_letters=('t',) + historic_parameter.dims.letters[1:])
-    historic_scaler = get_new_array(dims=dims, dim_letters=('h',)+scaler.dims.letters[1:])
-    historic_share = get_new_array(dims=dims, dim_letters=historic_parameter.dims.letters)
-
-    historic_scaler[...] = scaler * transform_t_thist
-    historic_scaler[...] =  historic_scaler.maximum(1e-10)  # have a minimum value to avoid division by zero
-
-    historic_share[...] = historic_parameter / historic_scaler
+    future_parameter.values[:n_historic] = historic_parameter.values
 
-    scale_factor = (0.3* historic_share[...][scaling_year-1] +
-                    0.25 * historic_share[...][scaling_year-2] +
-                    0.2 * historic_share[...][scaling_year-3] +
-                    0.15 * historic_share[...][scaling_year-4] +
-                    0.1 * historic_share[...][scaling_year-5])
+    # add missing dimensions to scaler values
+    scaler_values = scaler.values
+    for i, letter in enumerate(historic_parameter.dims.letters[1:],start=1):  # first dim is h which should not be ammended
+        if letter not in scaler.dims.letters:
+            scaler_values = np.expand_dims(scaler_values, axis=i)
 
-    future_parameter[...] = scaler * scale_factor
+    extrapolation = LinearExtrapolation(data_to_extrapolate=historic_parameter.values,
+                                        extrapolate_from=scaler_values)
+    future_parameter.values = extrapolation.predict()
```

Comment:
Does this work with `future_parameter[...] = extrapolation.predict()` ?

Reply:
Hm, the way I understood the code structure now is that within data_extrapolations we work with np arrays, whereas in general we work with NDArrays (or in the future XArrays). Jakob adapted this now slightly using the NDArray 'set_values' method instead of changing the values attribute as I did here which seems a bit cleaner. Do you think it's fine like that? Otherwise we could also make sure that the data extrapolations return an ND array directly.

Reply:
Okay, yes, fine for me for now. At some point later we can/should revisit how the extrapolations are being used and try to improve code structure in the data_translations and/or move it into new/existing classes, but I would leave this until we have other usecases (i.e. when I get around to making the plastics model).